diff --git a/node_modules/onnxruntime-react-native/android/build.gradle b/node_modules/onnxruntime-react-native/android/build.gradle
index 41b4359..a1c2e3f 100644
--- a/node_modules/onnxruntime-react-native/android/build.gradle
+++ b/node_modules/onnxruntime-react-native/android/build.gradle
@@ -84,17 +84,19 @@
         cppFlags "-O2 -frtti -fexceptions -Wall -Wno-unused-variable -fstack-protector-all"
         if (REACT_NATIVE_MINOR_VERSION >= 71) {
           // fabricjni required c++_shared
-          arguments "-DANDROID_STL=c++_shared",
-            "-DNODE_MODULES_DIR=${nodeModules}",
+          arguments "-DCMAKE_SHARED_LINKER_FLAGS=-Wl,-z,max-page-size=16384",
+            "-DANDROID_STL=c++_shared",
+            "-DNODE_MODULES_DIR=${nodeModules}",
             "-DORT_EXTENSIONS_ENABLED=${ortExtensionsEnabled}",
             "-DREACT_NATIVE_VERSION=${REACT_NATIVE_VERSION}",
             "-DUSE_QNN=${useQnn}",
             "-DUSE_NNAPI=${!useQnn}"
         } else {
-          arguments "-DNODE_MODULES_DIR=${nodeModules}",
+          arguments "-DCMAKE_SHARED_LINKER_FLAGS=-Wl,-z,max-page-size=16384",
+            "-DNODE_MODULES_DIR=${nodeModules}",
             "-DORT_EXTENSIONS_ENABLED=${ortExtensionsEnabled}",
             "-DREACT_NATIVE_VERSION=${REACT_NATIVE_VERSION}",
             "-DUSE_QNN=${useQnn}",
             "-DUSE_NNAPI=${!useQnn}"
         }
         abiFilters (*reactNativeArchitectures())
@@ -247,7 +249,7 @@ dependencies {
     extractLibs "com.microsoft.onnxruntime:onnxruntime-android:latest.integration@aar"
   }
 
-  if (VersionNumber.parse(REACT_NATIVE_VERSION) < VersionNumber.parse("0.71")) {
+  if (REACT_NATIVE_MINOR_VERSION < 71) {
     extractLibs "com.facebook.fbjni:fbjni:+:headers"
     extractLibs "com.facebook.fbjni:fbjni:+"
   }
